File: dynamic_form_gen/src/main/java/com/ihsinformatics/dynamicformsgenerator/utils/AppUtility.java
package com.ihsinformatics.dynamicformsgenerator.utils;

import android.content.res.Resources;
import android.util.TypedValue;

import java.util.Date;
import java.util.concurrent.TimeUnit;

/**
 * Created by Nabil shafi on 6/14/2018.
 */

public class AppUtility {

    /**
     * Converting dp to pixel
     */
    public static int dpToPx(Resources r, int dp) {
        return Math.round(TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, r.getDisplayMetrics()));
    }

    public static long getDifferenceDays(Date d1, Date d2) {
        long diff = d2.getTime() - d1.getTime();
        return TimeUnit.DAYS.convert(diff, TimeUnit.MILLISECONDS);
    }
}

File: app/src/main/java/ihsinformatics/com/hydra_mobile/utils/AppUtility.kt
package ihsinformatics.com.hydra_mobile.utils

import android.content.Context
import android.content.DialogInterface
import android.content.res.Resources
import android.util.TypedValue
import androidx.appcompat.app.AlertDialog
import java.util.*
import java.util.concurrent.TimeUnit

/**
 *
 *
 */

class AppUtility {

    companion object {
        /**
         * Converting dp to pixel
         */
        fun dpToPx(r: Resources, dp: Int): Int {
            return Math.round(TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp.toFloat(), r.displayMetrics))
        }

        fun getDifferenceDays(d1: Date, d2: Date): Long {
            val diff = d2.time - d1.time
            return TimeUnit.DAYS.convert(diff, TimeUnit.MILLISECONDS)
        }

        fun showMessageOKCancel(context: Context, message: String, okListener: DialogInterface.OnClickListener) {
            AlertDialog.Builder(context)
                .setMessage(message)
                .setPositiveButton("OK", okListener)
                .setNegativeButton("Cancel", null)
                .create()
                .show()
        }
    }

}